### DISCUSSION TITLE
script mount window , import script local  window.aaa  is undefined

### DISCUSSION THREAD

**Discussion by loubingzhuxue** [nuxt/nuxt#17165] [Questions]
### Environment

Operating System: Linux
Node Version: v17.2.0
Nuxt Version: 3.0.0-rc.6
Package Manager: npm@8.1.4
Bundler: Vite
User Config: alias, css
Runtime Modules: -
Build Modules: -

### Reproduction

https://stackblitz.com/edit/github-hb9skg?file=app.vue

script mount window, import script local undefined

### Describe the bug

script mount window, first link to script.vue , import script local window.aaa is undefined 。

what is the solution ？

//  /public/script.js
```
window.aaa = 12345
```

// /pages/index.vue
```
<template>
    <NuxtLink class="link" to="/script">Next Script</NuxtLink>
</template>
```

// /pages/script.vue
```
<script setup>
useHead({
  script: [
    {
      children: `console.log('Hello world script !')`,
      body: true,
    },
    {
      src: "/script.js",
      body: true,
    },
    {
      src: "https://g.alicdn.com/AWSC/AWSC/awsc.js",
      body: true,
    },
  ],
});
onMounted(() => {
  // first link to script.vue window.aaa is undefined
  console.log(window.aaa);  // undefined
  console.log(window.AWSC); // undefined
});
</script>
```

**Comment by huang-julien** [comment]
it seems that the onMounted() defined in the component is triggered immediatly after usehead.
![image](https://user-images.githubusercontent.com/63512348/182129967-d3b897a8-aa89-4573-a48e-a528ae6366bf.png)
Try to wait for a few millisecond (not the ideal solution, but it does work)
```ts
onMounted(async () => {
  await setTimeout(() => {
    console.log('Script', nameRef.value, document.querySelector('#script'));
    // first link to script.vue window.aaa is undefined
    console.log(window.aaa); // undefined
    console.log(window.AWSC); // undefined
  }, 10);
});
```

or using an event with onload for each scripts ?
